<?xml version="1.0" encoding="UTF-8"?>
<ui version="4.0">
 <class>MainWindow</class>
 <widget class="QMainWindow" name="MainWindow">
  <property name="geometry">
   <rect>
    <x>0</x>
    <y>0</y>
    <width>500</width>
    <height>410</height>
   </rect>
  </property>
  <property name="windowTitle">
   <string>Qt多线程范例-保存文件副本-星夜之夏</string>
  </property>
  <widget class="QWidget" name="centralWidget">
   <widget class="QGroupBox" name="grpOperation1">
    <property name="geometry">
     <rect>
      <x>10</x>
      <y>10</y>
      <width>481</width>
      <height>71</height>
     </rect>
    </property>
    <property name="title">
     <string>多线程复制（方法：重写run函数）</string>
    </property>
    <property name="checkable">
     <bool>true</bool>
    </property>
    <widget class="QPushButton" name="btnCancel1">
     <property name="enabled">
      <bool>false</bool>
     </property>
     <property name="geometry">
      <rect>
       <x>300</x>
       <y>30</y>
       <width>100</width>
       <height>30</height>
      </rect>
     </property>
     <property name="text">
      <string>取消</string>
     </property>
    </widget>
    <widget class="QPushButton" name="btnCopy1">
     <property name="geometry">
      <rect>
       <x>70</x>
       <y>30</y>
       <width>100</width>
       <height>30</height>
      </rect>
     </property>
     <property name="text">
      <string>复制</string>
     </property>
    </widget>
   </widget>
   <widget class="QGroupBox" name="grpProgress">
    <property name="geometry">
     <rect>
      <x>10</x>
      <y>260</y>
      <width>481</width>
      <height>80</height>
     </rect>
    </property>
    <property name="title">
     <string>复制进度</string>
    </property>
    <widget class="QProgressBar" name="progressBar">
     <property name="geometry">
      <rect>
       <x>40</x>
       <y>30</y>
       <width>391</width>
       <height>30</height>
      </rect>
     </property>
     <property name="value">
      <number>0</number>
     </property>
    </widget>
    <widget class="QProgressBar" name="progressBar2">
     <property name="geometry">
      <rect>
       <x>40</x>
       <y>60</y>
       <width>391</width>
       <height>10</height>
      </rect>
     </property>
     <property name="maximum">
      <number>0</number>
     </property>
     <property name="value">
      <number>-1</number>
     </property>
    </widget>
   </widget>
   <widget class="QPushButton" name="btnQuit">
    <property name="geometry">
     <rect>
      <x>390</x>
      <y>350</y>
      <width>100</width>
      <height>30</height>
     </rect>
    </property>
    <property name="text">
     <string>退出</string>
    </property>
   </widget>
   <widget class="QGroupBox" name="grpOperation2">
    <property name="geometry">
     <rect>
      <x>10</x>
      <y>90</y>
      <width>481</width>
      <height>71</height>
     </rect>
    </property>
    <property name="title">
     <string>多线程复制（方法：MoveToThread）</string>
    </property>
    <property name="flat">
     <bool>false</bool>
    </property>
    <property name="checkable">
     <bool>true</bool>
    </property>
    <property name="checked">
     <bool>false</bool>
    </property>
    <widget class="QPushButton" name="btnCancel2">
     <property name="enabled">
      <bool>false</bool>
     </property>
     <property name="geometry">
      <rect>
       <x>300</x>
       <y>30</y>
       <width>100</width>
       <height>30</height>
      </rect>
     </property>
     <property name="text">
      <string>取消</string>
     </property>
    </widget>
    <widget class="QPushButton" name="btnCopy2">
     <property name="geometry">
      <rect>
       <x>70</x>
       <y>30</y>
       <width>100</width>
       <height>30</height>
      </rect>
     </property>
     <property name="text">
      <string>复制</string>
     </property>
    </widget>
   </widget>
   <widget class="QGroupBox" name="grpOperation3">
    <property name="geometry">
     <rect>
      <x>10</x>
      <y>180</y>
      <width>481</width>
      <height>71</height>
     </rect>
    </property>
    <property name="title">
     <string>单线程复制（观察下方进度条，并尝试点击取消，会发现卡住）</string>
    </property>
    <property name="flat">
     <bool>false</bool>
    </property>
    <property name="checkable">
     <bool>true</bool>
    </property>
    <property name="checked">
     <bool>false</bool>
    </property>
    <widget class="QPushButton" name="btnCancel3">
     <property name="enabled">
      <bool>false</bool>
     </property>
     <property name="geometry">
      <rect>
       <x>300</x>
       <y>30</y>
       <width>100</width>
       <height>30</height>
      </rect>
     </property>
     <property name="text">
      <string>取消</string>
     </property>
    </widget>
    <widget class="QPushButton" name="btnCopy3">
     <property name="geometry">
      <rect>
       <x>70</x>
       <y>30</y>
       <width>100</width>
       <height>30</height>
      </rect>
     </property>
     <property name="text">
      <string>复制</string>
     </property>
    </widget>
   </widget>
   <widget class="QLabel" name="lblCopyright">
    <property name="geometry">
     <rect>
      <x>10</x>
      <y>350</y>
      <width>300</width>
      <height>30</height>
     </rect>
    </property>
    <property name="text">
     <string>Copyright 星夜之夏</string>
    </property>
   </widget>
  </widget>
  <widget class="QStatusBar" name="statusBar"/>
 </widget>
 <layoutdefault spacing="6" margin="11"/>
 <resources/>
 <connections/>
</ui>
